vela-templates/definitions/internal/workflowstep/generate-jdbc-connection.cue

```diff
@@ -0,0 +1,45 @@
+import (
+	"vela/op"
+	"encoding/base64"
+)
+
+"generate-jdbc-connection": {
+	type: "workflow-step"
+	annotations: {}
+	labels: {
+		"ui-hidden": "true"
+	}
+	description: "Generate a JDBC connection based on Component of alibaba-rds"
+}
+template: {
+	output: op.#Read & {
+		value: {
+			apiVersion: "v1"
+			kind:       "Secret"
+			metadata: {
+				name: parameter.name
+				if parameter.namespace != _|_ {
+					namespace: parameter.namespace
+				}
+			}
+		}
+	}
+	dbHost:   op.#ConvertString & {bt: base64.Decode(null, output.value.data["DB_HOST"])}
+	dbPort:   op.#ConvertString & {bt: base64.Decode(null, output.value.data["DB_PORT"])}
+	dbName:   op.#ConvertString & {bt: base64.Decode(null, output.value.data["DB_NAME"])}
+	username: op.#ConvertString & {bt: base64.Decode(null, output.value.data["DB_USER"])}
+	password: op.#ConvertString & {bt: base64.Decode(null, output.value.data["DB_PASSWORD"])}
+
+	env: [
+		{name: "url", value:      "jdbc://" + dbHost.str + ":" + dbPort.str + "/" + dbName.str + "?characterEncoding=utf8&useSSL=false"},
```

Comment:
advised to make it a configurable parameter, `characterEncoding=utf8&useSSL=false` This is not suitable for all.

others: Can env be injected directly?

Reply:
Yes, you are right. Let's make it simple and clear at the very beginning. And wait and hear feedback from end-users. I will expose them to users after that. Thanks.